### ISSUE TITLE
`TypeIs` returns `Never` instead of intersection type

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#7647]
**Describe the bug**

A `TypeIs` function is currently not mimicing the behavior of `isinstance` and narrows type to `Never` if two types are not consistent.

Note that if the `isinstance` part in the sample snippet gets uncommented, the narrowed type becomes the correct "\<subclass of `int` and `str`\>".

**Code or Screenshots**

```python
from typing_extensions import TypeIs

def is_str(x: object, /) -> TypeIs[str]: ...

x: int = globals()[""]
#  if isinstance(x, str):
#      pass
if is_str(x):
    reveal_type(x)  # Never
```

**VS Code extension or command-line**

Pyright 1.1.357 CLI.

**Comment by Azureblade3808** [comment]
Will you please take a look at the inconsistence between whether there is an `if isinstance(...)` statement beforehand? It looks very suspicious to me.

**Comment by erictraut** [comment]
I'm reopening this issue and investigating further.

**Comment by erictraut** [comment]
I've decided to combine the logic for `TypeIs` and `isinstance` narrowing so they are consistent. That's likely the behavior that most pyright users expect. This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.361, which I just published.
